org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/InferenceContext18.java

```diff
@@ -2078,14 +2078,14 @@ private boolean findRPrimeAndResultingBounds(TypeBinding typeBinding, InferenceV
 		} /* else part: Otherwise, B2 is the same as B1.*/
 		return true;
 	}
-	private TypeBinding deriveTPrime(TypeBinding candidateT, InferenceVariable[] alphas, TypeBinding typeBinding) {
+	private TypeBinding deriveTPrime(RecordPattern recordPattern, TypeBinding candidateT, InferenceVariable[] alphas, TypeBinding typeBinding) {
 		ParameterizedTypeBinding parameterizedType = null;
 		TypeBinding tPrime = null;
 		if (candidateT.isParameterizedType()) {
 			parameterizedType = InferenceContext18.parameterizedWithWildcard(candidateT);
 		}
 		if (parameterizedType != null && parameterizedType.arguments != null) {
-			TypeBinding[] arguments = parameterizedType.arguments;
+			TypeBinding[] arguments = parameterizedType.capture(this.scope, recordPattern.sourceStart, recordPattern.sourceEnd).arguments;
```

Comment:
refresh from the spec saying "_let Tc be the result of capture conversion (5.1.10) applied to T_"

Reply:
+1